src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -67,26 +169,23 @@ internal set
             }
         }
 
-        internal TimeSpan OperationTimeout
-        {
-            get { return this.operationTimeout; }
-        }
+        public virtual string Path { get; private set; }
+
+        public DateTime LockedUntilUtc { get; protected set; }
+
+        public string SessionId { get; protected set; }
+
+        internal TimeSpan OperationTimeout { get; private set; }
 
         internal MessagingEntityType? EntityType { get; set; }
```

Comment:
>set [](start = 56, length = 3)

Updating the entityType after creating the link doesnt work, we should mark the `set` private.